meilisearch/src/routes/indexes/mod.rs

```diff
@@ -90,13 +90,14 @@ pub async fn list_indexes(
     paginate: AwebQueryParameter<ListIndexes, DeserrQueryParamError>,
 ) -> Result<HttpResponse, ResponseError> {
     let search_rules = &index_scheduler.filters().search_rules;
-    let indexes: Vec<_> = index_scheduler.indexes()?;
-    let indexes = indexes
-        .into_iter()
-        .filter(|(name, _)| search_rules.is_index_authorized(name))
-        .map(|(name, index)| IndexView::new(name, &index))
-        .collect::<Result<Vec<_>, _>>()?;
-
+    let indexes: Vec<Option<IndexView>> =
+        index_scheduler.try_for_each_index(|uid, index| -> Result<Option<IndexView>, _> {
+            if !search_rules.is_index_authorized(uid) {
+                return Ok(None);
+            }
+            Ok(Some(IndexView::new(uid.to_string(), index)?))
+        })?;
+    let indexes: Vec<_> = indexes.into_iter().flatten().collect();
```

Comment:
Aren't we collecting the opened indexes in a single `Vec` here?

Reply:
no the type is `IndexView` which doesn't keep a reference to the index (only uses it in its constructor). I will add a comment stating so.

Reply:
done, see https://github.com/meilisearch/meilisearch/pull/3331/commits/535fdc1bc0a943a209e6409b6030d1852fc747f8